src/main/scala/com.snowplowanalytics.sauna/observers/AmazonKinesisObserver.scala

```diff
@@ -0,0 +1,121 @@
+
+// sauna
+import Observer.KinesisRecordReceived
+
+class AmazonKinesisObserver(streamName: String, kclConfig: KinesisClientLibConfiguration) extends Actor with Observer {
+  override def preStart: Unit = {
+    super.preStart()
+    notifyLogger("Started Kinesis Observer")
+
+    /**
+     * [[KinesisRecordReader]] instance that will return processed records
+     * without any transformation.
+     */
+    implicit object ARecordReader extends KinesisRecordReader[Record] {
+      override def apply(r: Record): Record = {
+        r
+      }
+    }
+
+    /**
+     * Start a worker thread that will notify the mediator about processed
+     * Kinesis records.
+     */
+    KCLWorkerRunner(kclConfig).runAsyncSingleRecordProcessor[Record](1 minute) { record: Record =>
+      Future {
+        notifyLogger(s"Received Kinesis Record from $streamName")
+        context.parent ! KinesisRecordReceived(streamName, record.getSequenceNumber, record.getData, self)
+      }
+    }
+  }
+
+  def receive: Receive = {
+    case _ =>
+  }
+}
+
+object AmazonKinesisObserver {
+  def props(streamName: String, kclConfig: KinesisClientLibConfiguration): Props =
+    Props(new AmazonKinesisObserver(streamName, kclConfig))
+
+  def props(config: AmazonKinesisConfig_1_0_0): Props = {
+
+    // AWS configuration. Safe to throw exception on initialization
+    val credentials = new BasicAWSCredentials(
+      config.parameters.aws.accessKeyId,
+      config.parameters.aws.secretAccessKey)
+
+    class KinesisCredentialsProvider(credentials: BasicAWSCredentials) extends AWSCredentialsProvider {
+      def refresh(): Unit = {}
+
+      def getCredentials: BasicAWSCredentials = credentials
+    }
+
+    val credentialsProvider = new KinesisCredentialsProvider(credentials)
+
+    var kclConfiguration = KCLConfiguration(
+      config.id,
+      config.parameters.kinesis.streamName,
+      credentialsProvider,
+      credentialsProvider,
+      credentialsProvider
+    )
+      .withRegionName(config.parameters.kinesis.region)
+      .withMaxRecords(config.parameters.kinesis.maxRecords)
+
+    config.parameters.kinesis.initialPosition match {
+      case ShardIteratorType_1_0_0.LATEST =>
+        kclConfiguration = kclConfiguration.withInitialPositionInStream(InitialPositionInStream.LATEST)
+      case ShardIteratorType_1_0_0.TRIM_HORIZON =>
+        kclConfiguration = kclConfiguration.withInitialPositionInStream(InitialPositionInStream.TRIM_HORIZON)
+      case ShardIteratorType_1_0_0.AT_TIMESTAMP =>
+        (for {
+          modifiers <- config.parameters.kinesis.initialPositionModifiers
+          timestamp <- modifiers.timestamp
+        } yield timestamp) match {
+          case Some(ts) =>
+            kclConfiguration = kclConfiguration
+              .withInitialPositionInStream(InitialPositionInStream.AT_TIMESTAMP)
+              .withTimestampAtInitialPositionInStream(new Date(ts))
+          case None =>
+            Props.empty
+        }
+    }
```

Comment:
I'd suggest refactoring the `kclConfiguration` building into its own function to make things clearer, I'm not so sure about the `Props.empty` above